### PR TITLE
LLClient: Add String flavored setEntity

### PR DESCRIPTION
**Pull request by nik9000**
Adds `Request#setEntity(String)` which short circuits the process of
sending a json string which is super common.

### CODE REVIEW

**Comment by nik9000** [comment]
@javanna this follows up on some comments you've made on my recent PRs to use the `Request` flavored methods in the http client.

I think this extra method would be pretty useful. I'm not really sold on a `byte[]` version of the method though because it feels like we're adding more and more flavors for less and less common things. So I figure I'd propose this change and see where it goes.

File: modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteRequestBuilders.java [hunk 1]
```diff
@@ -151,8 +149,7 @@ static Request initialSearch(SearchRequest searchRequest, BytesReference query,
             }
 
             entity.endObject();
-            BytesRef bytes = BytesReference.bytes(entity).toBytesRef();
-            request.setEntity(new ByteArrayEntity(bytes.bytes, bytes.offset, bytes.length, ContentType.APPLICATION_JSON));
+            request.setEntity(Strings.toString(entity));
```
Comment (dadoonet) [inline modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteRequestBuilders.java]: Should we support as well the method for bytes?

File: modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteRequestBuilders.java [hunk 2]
```diff
@@ -199,15 +196,15 @@ static Request scroll(String scroll, TimeValue keepAlive, Version remoteVersion)
...
+            request.setEntity(new NStringEntity(scroll, ContentType.TEXT_PLAIN));
             return request;
         }
 
         try (XContentBuilder entity = JsonXContent.contentBuilder()) {
             entity.startObject()
                     .field("scroll_id", scroll)
                 .endObject();
-            request.setEntity(new StringEntity(Strings.toString(entity), ContentType.APPLICATION_JSON));
+            request.setEntity(Strings.toString(entity));
```
Comment (dadoonet) [inline modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteRequestBuilders.java]: Should we support the method for XContent as well?

**Review by dadoonet — COMMENTED** [review]
I like that change! 😉
I left 2 comments.

File: modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteRequestBuilders.java (hunk 1 above)
Comment (dadoonet) [reply modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteRequestBuilders.java]: Ha! Just saw your other comment.

File: client/rest/src/main/java/org/elasticsearch/client/Request.java [hunk 3]
```diff
@@ -103,6 +105,17 @@ public void setEntity(HttpEntity entity) {
...
     }
 
+    /**
+     * Set the body of the request to a string. If not set or set to
+     * {@code null} then no body is sent with the request. The
+     * {@code Content-Type} will be sent as {@code application/json}.
+     * If you need a different content type then use
+     * {@link #setEntity(HttpEntity)}.
+     */
+    public void setEntity(String entity) {
```
Comment (javanna) [inline client/rest/src/main/java/org/elasticsearch/client/Request.java]: nit: call it setJsonEntity?

File: modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteRequestBuilders.java (hunk 2 above)
Comment (javanna) [reply modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteRequestBuilders.java]: low-level client doesn't depend on xcontent, we can't do that

File: modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteRequestBuilders.java (hunk 1 above)
Comment (javanna) [reply modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteRequestBuilders.java]: on this one I am on the fence. It is odd that we have to go from bytes to string here. I would rather leave what you had, or add a method to set a byte array, even if we are the only ones using it.

File: client/rest/src/main/java/org/elasticsearch/client/Request.java [hunk 4]
```diff
@@ -103,6 +105,17 @@ public void setEntity(HttpEntity entity) {
...
+        setEntity(entity == null ? null : new NStringEntity(entity, ContentType.APPLICATION_JSON));
```
Comment (javanna) [inline client/rest/src/main/java/org/elasticsearch/client/Request.java]: can we add a small unit test for this method?

**Review by javanna — APPROVED** [review]
left a few nits, LGTM otherwise. thanks @nik9000 !

**Comment by nik9000** [comment]
Yikes! I'm not sure how I let the failure through....

**Comment by nik9000** [comment]
Thanks for having a look @javanna and @dadoonet! I've merged and am cherry-picking back to 6.x now.

**Pull request merged**